### PR TITLE
DE-18299: Added UI component for file upload.

### PR DESCRIPTION
**Pull request by sushil-meher-nice**

### CODE REVIEW

File: src-php/Controls/FileUpload/FileUpload.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+    public function __construct(
+        string $id,
+        string $name,
+        string $value = '',
+        string $placeholder = '',
+        string $accept = '',
+        bool $isMultiple = false,
+        bool $isError = false
```
Comment (mksada) [inline src-php/Controls/FileUpload/FileUpload.php:54]: [O]
```suggestion
        bool $hasError = false
```

File: src-php/Controls/FileUpload/FileUpload.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    public function render(): string
+    {
+        $accept = $this->accept !== '' ?
+            sprintf('accept="%s" ', StringEscaper::escapeHtmlAttribute($this->accept))
+            : '';
+
+        $multiple = $this->isMultiple ? 'multiple ' : '';
+
+        $errorClass = $this->isError ? ' Input__Error' : '';
```
Comment (mksada) [inline src-php/Controls/FileUpload/FileUpload.php:74]: [O] I would put Input__Error into constant

File: src-php/Controls/FileUpload/FileUpload.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    private $isError;
+
+
+    public function __construct(
+        string $id,
+        string $name,
+        string $value = '',
+        string $placeholder = '',
+        string $accept = '',
```
Comment (mksada) [inline src-php/Controls/FileUpload/FileUpload.php:52]: [O]
```suggestion
        string $acceptAttributeValue = '',
```

**Review by mksada — APPROVED** [review]

File: src-php/Controls/FileUpload/FileUpload.php (hunk 3 above)
Comment (sushil-meher-nice) [reply src-php/Controls/FileUpload/FileUpload.php]: Done.

File: src-php/Controls/FileUpload/FileUpload.php (hunk 1 above)
Comment (sushil-meher-nice) [reply src-php/Controls/FileUpload/FileUpload.php:54]: Keeping as it is in order to be consistent with other components.

File: src-php/Controls/FileUpload/FileUpload.php (hunk 2 above)
Comment (sushil-meher-nice) [reply src-php/Controls/FileUpload/FileUpload.php:74]: Keeping as it is in order to be consistent with other components.

**Review by roshan-deshmukh-nice — APPROVED** [review]

**Pull request merged**